File: src/main/java/com/caicai/swe/entity/Category.java
package com.caicai.swe.entity;

import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 分类实体
 */
@Data
public class Category implements Serializable {
    private static final long serialVersionUID = 1L;

    private Long id;
    private String name;
    private Long parentId;
    private Integer sort;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: frontend/src/api/category.ts
import request from '@/utils/request'

export interface CategoryVO {
  id: number
  name: string
  parentId: number
  sort: number
  createTime: string
}

export const categoryApi = {
  // 获取所有分类
  getAll: () => request.get<CategoryVO[]>('/category/list')
}